external_api/api_solargis.py

```diff
@@ -0,0 +1,271 @@
+import requests
+
+import datetime
+
+import xmlschema
+
+from typing import NamedTuple
+
+from conf.log import logger
+
+from plantmonitor.utils import rfc3336todt
+
+class Site(NamedTuple):
+    id: int
+    name: str
+    peak_power_w: int
+    latitude: float
+    longitude: float
+
+class ApiSolargis:
+
+    def __init__(self, api_base_url, api_key):
+
+        self.api_base_url = api_base_url
+        self.api_request_url = "{}/{}".format(api_base_url, "datadelivery/request")
+        self.api_key = api_key
+        self.api_name = 'solargis'
+
+        self.api_key_params = {'key': api_key}
+
+        # TODO set from database wither via Pony, sqlalchemy or else if we stick with a satellite api
+        # {plant_id: (lat, long)}
+
+        self.sites = {
+            3: Site(
+                id=3,
+                name='Fontivsolar',
+                latitude=40.932389,
+                longitude=-4.968694,
+                peak_power_w=990,
+            ),
+            9: Site(
+                id=9,
+                name='Picanya',
+                latitude=39.440722,
+                longitude=-0.428722,
+                peak_power_w=335,
+            ),
+            22: Site(
+                id=22,
+                name='Llanillos',
+                latitude=37.504330,
+                longitude=-3.236476,
+                peak_power_w=3820,
+            ),
+        }
+
+    def create_xsd_schema(self):
+        self.schema = xmlschema.XMLSchema('data/ws-data.xsd')
+
+    def get_arbitrary_payload(self, xml_request_content):
+        headers = {'Content-Type' : 'application/xml'}
+
+        try:
+            response = requests.post(
+                self.api_request_url,
+                params=self.api_key_params,
+                headers=headers,
+                data=xml_request_content.encode('utf8'),
+                timeout=30
+            )
+        except requests.Timeout:
```

Comment:
Si volem controlar totes les excepcions falta ConnectionError i TooManyRedirects (no sé quin codi http els pertoca)

except requests.exceptions.RequestException:

ja les estaríem capturant les tres per no haver de fer 3 excepts

https://docs.python-requests.org/en/latest/user/quickstart/#errors-and-exceptions